cmd/simulator/load/loader.go

```diff
@@ -124,6 +141,31 @@ func ExecuteLoader(ctx context.Context, config config.Config) error {
 		})
 	}
 
+	go func(ctx context.Context) {
+		// Start a prometheus server to expose individual tx metrics
+		server := &http.Server{
+			Addr: MetricsPort,
+		}
+
+		go func() {
+			defer func() {
+				if err := server.Shutdown(ctx); err != nil {
+					log.Error("Metrics server error: %v", err)
+				}
+				log.Info("Received a SIGINT signal: Gracefully shutting down metrics server")
+			}()
+			sigChan := make(chan os.Signal, 1)
+			signal.Notify(sigChan, syscall.SIGINT)
+			<-sigChan
+		}()
+
+		http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{Registry: reg}))
+		log.Info(fmt.Sprintf("Metrics Server: localhost%s/metrics", MetricsPort))
+		if err := server.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+			log.Error("Metrics server error: %v", err)
+		}
+	}(ctx)
```

Comment:
How can we make this simpler?

Reply:
Simplified this by moving this to a new function, improving readability by adding comments,  and moving all var declarations to the beginning of the function. I tried to mimic the avalanchego code [here. ](https://github.com/ava-labs/avalanchego/blob/master/vms/rpcchainvm/vm.go#L35-L70)

Reply:
Nice, this is 10x clearer